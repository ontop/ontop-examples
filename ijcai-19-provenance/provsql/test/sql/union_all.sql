\set ECHO none
SET search_path TO public,provsql;

CREATE TABLE union_all_result AS
SELECT *,formula(provenance(),'personnel_name') FROM (
  SELECT classification FROM personnel WHERE city='Paris'
  UNION ALL
  SELECT classification FROM personnel
) t;

SELECT remove_provenance('union_all_result');
SELECT * FROM union_all_result ORDER BY classification;
DROP TABLE union_all_result;

CREATE TABLE union_all_result AS
  SELECT * FROM personnel 
  UNION ALL
  SELECT * FROM personnel;

SELECT remove_provenance('union_all_result');
SELECT * FROM union_all_result ORDER BY id;
DROP TABLE union_all_result;
