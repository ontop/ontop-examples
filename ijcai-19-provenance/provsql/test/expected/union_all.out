\set ECHO none
 remove_provenance 
-------------------
 
(1 row)

 classification | formula  
----------------+----------
 unclassified   | John
 restricted     | Paul
 restricted     | Nancy
 restricted     | Nancy
 confidential   | Dave
 confidential   | Dave
 secret         | Ellen
 secret         | Susan
 top_secret     | Magdalen
 top_secret     | Magdalen
(10 rows)

 remove_provenance 
-------------------
 
(1 row)

 id |   name   |   position   |   city   | classification 
----+----------+--------------+----------+----------------
  1 | John     | Director     | New York | unclassified
  1 | John     | Director     | New York | unclassified
  2 | Paul     | Janitor      | New York | restricted
  2 | Paul     | Janitor      | New York | restricted
  3 | Dave     | Analyst      | Paris    | confidential
  3 | Dave     | Analyst      | Paris    | confidential
  4 | Ellen    | Field agent  | Berlin   | secret
  4 | Ellen    | Field agent  | Berlin   | secret
  5 | Magdalen | Double agent | Paris    | top_secret
  5 | Magdalen | Double agent | Paris    | top_secret
  6 | Nancy    | HR           | Paris    | restricted
  6 | Nancy    | HR           | Paris    | restricted
  7 | Susan    | Analyst      | Berlin   | secret
  7 | Susan    | Analyst      | Berlin   | secret
(14 rows)

